test/module/test/framework/integration_framework/port_guard_test.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <thread>
+
+#include <gtest/gtest.h>
+#include <boost/thread/barrier.hpp>
+#include <boost/noncopyable.hpp>
+#include "framework/integration_framework/port_guard.hpp"
+
+using PortGuard = integration_framework::PortGuard;
+
+TEST(PortGuardTest, AllPortsGetUsedAndNoOverlaps) {
+  static constexpr size_t kNumClients = 10; // first-run clients
+  static constexpr size_t kNumClientsKilled = 3; // then these die
+  static constexpr size_t kNumClientsResurrected = 5; // then these are added
+  static constexpr std::pair<size_t, size_t> kRange = {123, 456};
```

Comment:
Would be nice to have exact amount of available ports as the required number, since it's a better corner case

Reply:
Just tried with the whole port range - it was running for 1 min 16 sec. Too long, I think.

Reply:
Maybe reduce the range then?